### PR TITLE
Remove daml-on-sql

### PR DESCRIPTION
**Pull request by mziolekda**
CHANGELOG_BEGIN
Remove daml-on-sql
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by SamirTalwar-DA** [comment]
There are references to Daml on SQL in:

* _ci/cron/src/Docs.hs_
* _docs/source/ops/connect/helm.rst_
* _ledger/sandbox-on-x/README.md_

File: ledger/daml-on-sql/README.rst [hunk 1]
```diff
@@ -1,937 +0,0 @@
-.. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
```
Comment (SamirTalwar-DA) [inline ledger/daml-on-sql/README.rst:1]: Much of this document is quite general and not documented anywhere else. I believe we need to move large parts of it to the general documentation.

**Comment by mziolekda** [comment]
@SamirTalwar-DA Thanks for being really thorough

- ci/cron/src/Docs.hs -> this refers to past versions of the documents and needs to stay
- docs/source/ops/connect/helm.rst -> this needs to be redirected to canton sandbox as part of helm chart rework
- ledger/sandbox-on-x/README.md -> this is a comment that talks about sandbox-on-x being a replacement for daml-on-sql so very much still relevant

**Review by gerolf-da — APPROVED** [review]
🔥 🔥 🔥

**Review by SamirTalwar-DA — APPROVED** [review]

**Pull request merged**